### PR TITLE
Resource for VN-439

### PR DESCRIPTION
**Pull request by votuanphong1983**
Resource for ticket VN-439

### CODE REVIEW

File: cs/index.json [hunk 1]
```diff
@@ -97,7 +97,10 @@
     "error_cannot_connect_pool": "Nelze se připojit k poolu ({{poolinfo}}). Prosíme, zkontrolujte vaše připojení a ujistěte se, že port {{port}} není ve firewallu blokován.",
     "miner_stopped_cannot_connect_stats_port": "Port '{{port}}' je používán. Ujistěte se, že žádná další aplikace zrovna nepoužívá port {{port}}.",
     "miner_stop_unexpected": "Miner `{{type}}` byl neočekávaně zastaven",
-    "not_support_system": "System can not mine. Maybe RAM is not enough or your preferred algorithms are not suppored. Please check on 'Configure-Preferred Algorithms' menu. Minable algorithms list [{{algorithms}}]"
+    "not_support_system": "System can not mine. Maybe RAM is not enough or your preferred algorithms are not suppored. Please check on 'Configure-Preferred Algorithms' menu. Minable algorithms list [{{algorithms}}]",
```
Comment (huy2nhan) [inline cs/index.json]: I think saying `preferred algorithms are not suppored` sounds a bit odd, and `cannot` is more usual than `can not`. My suggestion:
```
Device is not supported or out of memory. Please check on 'Configure-Preferred Algorithms' menu. Minable algorithms: [{{algorithms}}]
```

File: cs/index.json [hunk 2]
```diff
@@ -97,7 +97,10 @@
...
+    "validate_fail_out_of_memory": "RAM is not enough. RAM must be at least {{ram}} MB",
```
Comment (huy2nhan) [inline cs/index.json]: RAM can be confused as machine RAM, should use VRAM instead or stats it as `device memory`?. My suggestion:
```
Insufficient device memory. VRAM must be at least {{vram}} MB."
```

File: cs/index.json [hunk 3]
```diff
@@ -97,7 +97,10 @@
...
+    "validate_fail_unsupported_device": "Unsupported device",
+    "predict_dag_expire_date": "DAG size of {{crypto}} is {{dagSize}} MB. You RAM({{curRAM}} MB) can mine to {{expireAt}}"
```
Comment (huy2nhan) [inline cs/index.json]: `You RAM({{curRAM}} MB) can mine to {{expireAt}}` is a bit odd and space missing. My suggestion:
```
DAG size of {{crypto}} is {{dagSize}} MB. Your device with {{curRAM}} MB can mine until {{expireAt}}
```

**Review by huy2nhan — DISMISSED** [review]

File: zh/index.json [hunk 4]
```diff
@@ -97,7 +97,10 @@
     "error_cannot_connect_pool": "無法連接礦池({{poolinfo}})。請檢查您的連接和防火牆以確保端口{{port}}未被阻擋。",
     "miner_stopped_cannot_connect_stats_port": "端口{{port}}已經在使用中。請檢查您的系統並確定沒有程式占用端口{{port}}。",
     "miner_stop_unexpected": "挖礦程式 `{{type}}` 意外停止了",
-    "not_support_system": "無法採礦。 也許你的系統記憶體(RAM)不夠或者你的首選算法不受支持。 請檢查'首選的演算法'菜單。 支持的算法有[{{algorithms}}]"
+    "not_support_system": "無法採礦。 也許你的系統記憶體(RAM)不夠或者你的首選算法不受支持。 請檢查'首選的演算法'菜單。 支持的算法有[{{algorithms}}]",
+    "validate_fail_out_of_memory": "Insufficient device memory. VRAM must be at least {{vram}} MB.",
+    "validate_fail_unsupported_device": "Unsupported device",
+    "predict_dag_expire_date": "DAG size of {{crypto}} is {{dagSize}}. You RAM can mine to {{expireAt}}"
```
Comment (pandafox) [inline zh/index.json]:
```
"validate_fail_out_of_memory": "顯示卡內存不足。VRAM必須至少為{{vram}} MB。",
"validate_fail_unsupported_device": "不支持的硬體設備"，
"predict_dag_expire_date": "{{crypto}}的DAG大小為{{dagSize}} MB。您的顯示卡有({{curRAM}} MB) 可以挖掘到 {{expireAt}}”
```

File: cs/index.json [hunk 5]
```diff
@@ -97,7 +97,10 @@
     "error_cannot_connect_pool": "Nelze se připojit k poolu ({{poolinfo}}). Prosíme, zkontrolujte vaše připojení a ujistěte se, že port {{port}} není ve firewallu blokován.",
     "miner_stopped_cannot_connect_stats_port": "Port '{{port}}' je používán. Ujistěte se, že žádná další aplikace zrovna nepoužívá port {{port}}.",
     "miner_stop_unexpected": "Miner `{{type}}` byl neočekávaně zastaven",
-    "not_support_system": "System can not mine. Maybe RAM is not enough or your preferred algorithms are not suppored. Please check on 'Configure-Preferred Algorithms' menu. Minable algorithms list [{{algorithms}}]"
+    "not_support_system": "Device is not supported or out of memory. Please check on 'Configure-Preferred Algorithms' menu. Minable algorithms: [{{algorithms}}]",
+    "validate_fail_out_of_memory": "Insufficient device memory. VRAM must be at least {{vram}} MB.",
+    "validate_fail_unsupported_device": "Unsupported device",
+    "predict_dag_expire_date": "DAG size of {{crypto}} is {{dagSize}} MB. You RAM({{curRAM}} MB) can mine to {{expireAt}}"
```
Comment (pandafox) [inline cs/index.json]:
```
    "validate_fail_out_of_memory": "Insufficient Video device memory. VRAM must be at least {{vram}} MB.",
    "validate_fail_unsupported_device": "Unsupported device",
    "predict_dag_expire_date": "DAG size of {{crypto}} is {{dagSize}} MB. Your Video RAM ({{curRAM}} MB) can mine to {{expireAt}}"
```

**Review by pandafox — CHANGES_REQUESTED** [review]
Do we still need these changes?

**Review by pandafox — APPROVED** [review]

**Pull request merged**